### PR TITLE
Add routing

### PR DESCRIPTION
**Pull request by toshiakitakase**

### CODE REVIEW

File: frontend/packages/react-app/src/App.tsx [hunk 1]
```diff
@@ -1,25 +1,52 @@
...
+            component={SelectDistributersPage}
+          />
+          <Route
+            exact
+            path="/dashboard/distributer/:address"
+            component={NewDistributerPage}
+          />
+          <Route
+            path="/dashboard/distributer/:address/:campaignId"
+            component={NewDistributerPage}
```
Comment (wild-mouse) [inline frontend/packages/react-app/src/App.tsx:34]: Should component be like `DistributerDetailPage` but not `NewDistributerPage`?

File: frontend/packages/react-app/src/App.tsx [hunk 2]
```diff
@@ -1,25 +1,52 @@
 import React, { Component, useContext, createContext } from "react";
 import { BrowserRouter as Router, Route } from "react-router-dom";
-import TopPage from "./components/pages/TopPage";
```
Comment (wild-mouse) [inline frontend/packages/react-app/src/App.tsx:3]: The top page is not designed in Figma but it seems necessary as a landing page.

File: frontend/packages/react-app/src/App.tsx (hunk 1 above)
Comment (wild-mouse) [reply frontend/packages/react-app/src/App.tsx:34]: Or is this just a placeholder until implemented?

**Comment by wild-mouse** [comment]
I want mapping information between routes and Figma pages.

File: frontend/packages/react-app/src/App.tsx (hunk 1 above)
Comment (toshiakitakase) [reply frontend/packages/react-app/src/App.tsx:34]: > Should component be like `DistributerDetailPage` but not `NewDistributerPage`?

OK!

File: frontend/packages/react-app/src/App.tsx (hunk 2 above)
Comment (toshiakitakase) [reply frontend/packages/react-app/src/App.tsx:3]: LP was to be cut out separately, so it is not needed here.

**Comment by toshiakitakase** [comment]
I updated and renamed Figma to be the same as routing.

**Comment by wild-mouse** [comment]
> I updated and renamed Figma to be the same as routing.

Thanks!

File: frontend/packages/react-app/src/App.tsx [hunk 3]
```diff
@@ -1,25 +1,59 @@
...
-          <Route path="/dashboard" component={DashboardPage} />
-          <Route path="/token/:address" component={TokenDetailPage} />
-          <Route path="/audius-token" component={WithdrawAudiusPage} />
-          <Route path="/audius/:address" component={AudiusDisributionPage} />
+          <Route exact path="/" component={ExplorePage} />
+
+          {/* For Creator */}
+          <Route exact path="/dashboard" component={DashboardPage} />
+          <Route
+            path="/dashboard/:address"
```
Comment (wild-mouse) [inline frontend/packages/react-app/src/App.tsx:24]: `/dashboard/:tokenAddress`

File: frontend/packages/react-app/src/App.tsx [hunk 4]
```diff
@@ -1,25 +1,59 @@
...
+            component={ExternalTokenDetailPage}
+          />
+          <Route
+            path="/dashboard/distributers"
```
Comment (wild-mouse) [inline frontend/packages/react-app/src/App.tsx:28]: `/dashboard/:tokenAddress/distributers` (seems `tokenAddress` is necessary for history.push and sending transaction on next page)

File: frontend/packages/react-app/src/App.tsx [hunk 5]
```diff
@@ -1,25 +1,59 @@
...
+            component={SelectDistributersPage}
+          />
+          <Route
+            exact
+            path="/dashboard/distributer/:address"
```
Comment (wild-mouse) [inline frontend/packages/react-app/src/App.tsx:33]: `/dashboard/:tokenAddress/distributers/:distributerAddress`

File: frontend/packages/react-app/src/App.tsx [hunk 6]
```diff
@@ -1,25 +1,59 @@
...
+            component={CreateCampaignPage}
+          />
+          <Route
+            path="/dashboard/distributer/:address/:campaignId"
```
Comment (wild-mouse) [inline frontend/packages/react-app/src/App.tsx:37]: `/dashboard/:tokenAddress/distributers/:distributerAddress/campaigns/:campaignId(address)`

File: frontend/packages/react-app/src/App.tsx [hunk 7]
```diff
@@ -1,25 +1,59 @@
...
+            component={CampaignDetailPage}
+          />
+
+          {/* For Fan */}
+          <Route exact path="/explore" component={ExplorePage} />
+          <Route exact path="/explore/:address" component={TokenDetailPage} />
```
Comment (wild-mouse) [inline frontend/packages/react-app/src/App.tsx:43]: `/explore/:tokenAddress`

File: frontend/packages/react-app/src/App.tsx [hunk 8]
```diff
@@ -1,25 +1,59 @@
...
+          <Route path="/explore/:address/basic" component={TokenDetailPage} />
```
Comment (wild-mouse) [inline frontend/packages/react-app/src/App.tsx:44]: `/explore/:tokenAddress/basic`

File: frontend/packages/react-app/src/App.tsx [hunk 9]
```diff
@@ -1,25 +1,59 @@
...
+          <Route
+            path="/explore/:address/campaigns"
```
Comment (wild-mouse) [inline frontend/packages/react-app/src/App.tsx:46]: `/explore/:tokenAddress/campaigns`

File: frontend/packages/react-app/src/App.tsx [hunk 10]
```diff
@@ -1,25 +1,59 @@
...
+            component={TokenCampaignsPage}
+          />
+          <Route
+            path="/explore/:address/:campaignId"
```
Comment (wild-mouse) [inline frontend/packages/react-app/src/App.tsx:50]: `/explore/:tokenAddress/campaigns/:campaignId(address)`

File: frontend/packages/react-app/src/App.tsx [hunk 11]
```diff
@@ -1,25 +1,59 @@
...
+            component={TokenCampaignDetailPage}
+          />
+          <Route
+            path="/explore/:address/history"
```
Comment (wild-mouse) [inline frontend/packages/react-app/src/App.tsx:54]: `/explore/:tokenAddress/history`

File: frontend/packages/react-app/src/App.tsx [hunk 12]
```diff
@@ -1,25 +1,59 @@
...
       <Router>
         <div>
-          <Route exact path="/" component={TopPage} />
-          <Route path="/create" component={CreateTokenPage} />
-          <Route path="/explore" component={ExplorePage} />
-          <Route path="/dashboard" component={DashboardPage} />
-          <Route path="/token/:address" component={TokenDetailPage} />
-          <Route path="/audius-token" component={WithdrawAudiusPage} />
-          <Route path="/audius/:address" component={AudiusDisributionPage} />
+          <Route exact path="/" component={ExplorePage} />
```
Comment (wild-mouse) [inline frontend/packages/react-app/src/App.tsx:19]: We should standardize paths like the following:

**Comment by toshiakitakase** [comment]
Fixed!

**Review by wild-mouse — APPROVED** [review]
LGTM 🏃

**Pull request merged**